x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/failurestore/FailureStoreSecurityRestIT.java

```diff
@@ -1827,4 +2480,18 @@ private Tuple<String, String> getSingleDataAndFailureIndices(String dataStreamNa
         assertThat(indices.v2().size(), equalTo(1));
         return new Tuple<>(indices.v1().get(0), indices.v2().get(0));
     }
+
+    private void expectHasPrivileges(String user, String requestBody, String expectedResponse) throws IOException {
+        Request req = new Request("POST", "/_security/user/_has_privileges");
+        req.setJsonEntity(requestBody);
+        Response response = performRequestMaybeUsingApiKey(user, req);
```

Comment:
optional: we could also randomly switch here to use `run_as` to check privileges on behalf of other user.